src/components/Forms/SignIn/SignIn.js

```diff
@@ -0,0 +1,215 @@
+import './SignIn.scss';
+
+export class SignIn extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      showPassword: false,
+      errors: {},
+      isFormValid: true,
+    };
+  }
+
+  handleChange = (e) => {
+    const { name, value } = e.target;
+    const errors = {};
+    const [errorValue] = validate({
+      [name]: value,
+    }, name === 'password' ? passwordSchema : emailSchema);
+
+    errors[name] = errorValue || '';
+    this.setFormValidity(errors);
+    this.setState({
+      [name]: value,
+      errors,
+    });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { signIn, history } = this.props;
+    const {
+      email,
+      password,
+      errors,
+    } = this.state;
+    const formIsValid = this.validateForm();
+
+    if (!formIsValid) {
+      return this.setFormValidity(errors);
+    }
+    signIn({ email, password }, history);
+  }
+
+  validateForm = () => {
+    const errors = {};
+    const {
+      email,
+      password,
+    } = this.state;
+    const [emailError] = validate({ email }, emailSchema);
+    const [passwordError] = validate({ password }, passwordSchema);
+
+    errors.email = emailError || '';
+    errors.password = passwordError || '';
+    return this.setFormValidity(errors);
+  }
+
+  setFormValidity = (errors) => {
+    let valid = true;
+    Object.values(errors).forEach((value) => {
+      if (value.length > 0) {
+        valid = false;
+      }
+    });
+    this.setState({ isFormValid: valid });
+
+    return valid;
+  }
+
+  toggleVisibility = () => {
+    this.setState(prevState => ({
+      showPassword: !prevState.showPassword,
+    }));
+  }
+
+  resetModal = () => {
+    const { close } = this.props;
+    this.setState({
+      email: '',
+      password: '',
+      showPassword: false,
+      errors: {},
+      isFormValid: true,
+    });
+    close();
+  }
+
+  render() {
+    const {
+      email,
+      password,
+      showPassword,
+      errors,
+      isFormValid,
+    } = this.state;
+    const {
+      ui: {
+        loading,
+        modalOpen,
+        modal,
+      },
+    } = this.props;
+    const loader = <Loader type="BallTriangle" color="#fff" height={18} width={79} />;
+
+    return (
+      <Modal close={this.resetModal} open={modalOpen && modal === 'signin'}>
+        <div className="form">
+          <h3 className="form-header">Sign in</h3>
+          <p className="form-text">
+            Sign in to experience author’s haven great community of creative minds.
+            Follow your favourite authors and interact with the articles you love
+          </p>
+          <form style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center' }}>
+            <Input
+              name="email"
+              value={email}
+              type="email"
+              handleChange={this.handleChange}
+              placeholder="john.doe@foo.bar"
```

Comment:
I think it's cool we try something different from whats common as well. So I think the placeholders are cool. Good job on the research to come up with that as well.